File: Tarea1Unidad3Python/Producto.py
import random
from enum import Enum

class tipoProducto(Enum):
    LIMPIEZA = "Limpieza"
    BELLEZA = "Belleza"
    ELECTRODOMESTICOS = "Electrodomesticos"
    ALIMENTO = "Alimento"

class Producto:
    def __init__(self, nombre, tipo_producto, precio, stock, fecha_importacion):
        self.nombre = nombre
        self.precio = precio
        self.stock = stock
        self.fecha_importacion = fecha_importacion
        self.num_serie = self.generar_numero_serie()
        self.tipo_producto = tipo_producto
        self.comprado = False

    def get_nombre(self):
        return self.nombre

    def get_precio(self):
        return self.precio

    def get_stock(self):
        return self.stock

    def get_fecha_importacion(self):
        return self.fecha_importacion

    def get_num_serie(self):
        return self.num_serie

    def is_comprado(self):
        return self.comprado

    def set_comprado(self, comprado):
        self.comprado = comprado

    def get_tipo_producto(self):
        return self.tipo_producto

    def get_datos(self):
        return f"Nombre: {self.get_nombre()}\nPrecio: ${self.get_precio()}\nNúmero de serie: {self.get_num_serie()}\nFecha de importación: {self.get_fecha_importacion()}\nTipo de producto: {self.get_tipo_producto()}\nCantidad de productos: {self.get_stock()}"

    def set_stock(self, stock):
        self.stock = stock

    def generar_numero_serie(self):
        letra_nombre = self.get_nombre()[0]
        num = random.randint(100000, 999999)
        num_serie = f"{letra_nombre}-{num}"
        return num_serie

File: Tarea1Unidad3Java/src/Productos/Producto.java
package Productos;

import java.util.Random;
import Productos.helpers.tipoProducto;

public class Producto {
    private String nombre;
    private double precio;
    private int stock;
    private String fechaImportacion;
    private String numSerie;
    private tipoProducto tipoProducto;
    private boolean comprado;

    public Producto(String nombre, tipoProducto tipoProducto,  double precio, int stock, String fechaImportacion, String numSerie) {
        this.nombre = nombre;
        this.precio = precio;
        this.stock = stock;
        this.fechaImportacion = fechaImportacion;
        numSerie = numeroDeSerie();
        this.tipoProducto=tipoProducto;
        this.comprado=false;
    }

    public String getNombre() {
        return nombre;
    }

    public double getPrecio() {
        return precio;
    }

    public int getStock() {
        return stock;
    }

    public String getFechaImportacion() {
        return fechaImportacion;
    }

    public String getNumSerie() {
        return numSerie;
    }

    public boolean isComprado() {
        return comprado;
    }
    public void setComprado(boolean comprado) {
        this.comprado = comprado;
    }

    public Productos.helpers.tipoProducto getTipoProducto() {
        return tipoProducto;
    }

    public String getDatos(){
        System.out.printf("Nombre: %s \nPrecio: $%f \nNúmero de serie: %s \nFecha de importación: %s \nTipo de producto: %s \nCantidad de productos: %d\n", getNombre(), getPrecio(), getNumSerie(), getFechaImportacion(), getTipoProducto(), getStock());
        return("");
    }

    public void setStock(int stock) {
        this.stock = stock;
    }

    Random rd = new Random();
    public String numeroDeSerie(){
        char letraNombre=getNombre().charAt(0);
        int num = rd.nextInt(100000, 999999);
        numSerie=(letraNombre+"-"+String.valueOf(num));

        return numSerie;
    }
}